### PR TITLE
Add method to annotate stats to figures

### PR DESCRIPTION
**Pull request by CoryMartin-NOAA**
This closes issue #43 

A new method for `CreateMap` and `CreatePlot`, `add_stats_dict` will annotate a list of stats below the lower right corner of the figure.

Pass it a dictionary and it will annotate each key, value pair. See the example:

Screen Shot 2021-09-01 at 5 01 03 PM

### CODE REVIEW

**Comment by kevindougherty-noaa** [comment]
@CoryMartin-NOAA  Just curious, what if someone adds a lot of stats? Does it run into the x label?

**Comment by CoryMartin-NOAA** [comment]
@kevindougherty-noaa yeah probably, would you suggest it instead go in the very bottom right of the figure?

**Comment by kevindougherty-noaa** [comment]
@CoryMartin-NOAA Do you think it would be beneficial to move the colorbar to the bottom and then stats on the right side in a column? I prefer colorbar on the side, but that would probably make the figure you look neater.

**Comment by CoryMartin-NOAA** [comment]
@kevindougherty-noaa  Maybe this is a good time to do some sort of "literature review" and look at different sites and peer institutions and see what looks good, in terms of placement of color bar, metadata, titles, etc.

**Comment by CoryMartin-NOAA** [comment]
Screen Shot 2021-09-02 at 10 03 16 AM

Pardon the corner which is a xquartz/display artifact but I like them in the bottom like this. Thoughts?

**Comment by kevindougherty-noaa** [comment]
@CoryMartin-NOAA What do you think of this? I did some 'research' and a lot of plots have the colorbar on the bottom. ECMWF puts all the stats in the title, which I don't particularly like.

image

**Comment by CoryMartin-NOAA** [comment]
@kevindougherty-noaa my problem with the stats being on the right side is it looks like it introduces way too much empty whitespace into the PNG. I also do not want the stats in the title. Separate font sizes/locations make it easier to quickly glance to see things. Let's poll others. @RussTreadon-NOAA @ADCollard @emilyhcliu @aerorahul any opinions on where 'stats' such as n, min, max, should go on a figure?

**Comment by aerorahul** [comment]
> @kevindougherty-noaa my problem with the stats being on the right side is it looks like it introduces way too much empty whitespace into the PNG. I also do not want the stats in the title. Separate font sizes/locations make it easier to quickly glance to see things. Let's poll others. @RussTreadon-NOAA @ADCollard @emilyhcliu @aerorahul any opinions on where 'stats' such as n, min, max, should go on a figure?

For inspiration:
https://www.reddit.com/r/dataisbeautiful/

But that can get overwhelming quickly.

I actually like @emilyhcliu's graphics such as this:
Screen Shot 2021-09-03 at 2 25 47 PM

My suggestion would be to put the statistics under the colorbar and the colorbar label to the right of the colorbar.

**Comment by kevindougherty-noaa** [comment]
My latest commit with default settings produces this figure, replicating @emilyhcliu's approach and @aerorahul's suggestion:

```
d = {'Nobs': len(np.linspace(200, 300, 30)),
     'Mean': np.round(np.mean(np.linspace(200, 300, 30)), 4),
     'STD': np.round(np.std(np.linspace(200, 300, 30)), 4),
     'Min': np.min(np.linspace(200, 300, 30)),
     'Max': np.max(np.linspace(200, 300, 30))
    }

mymap.add_stats_dict(d, fontsize=12)
```

image

**Comment by CoryMartin-NOAA** [comment]
since @kevindougherty-noaa and I tag teamed this, will need a review from @aerorahul @RussTreadon-NOAA or @ADCollard

**Review by RussTreadon-NOAA — COMMENTED** [review]
My python level is insufficient to provide a detailed review.  One question, though.   It looks like `add_stats_dict` is identical in `create_map.py` and `create_plot.py`.  If true, can this common code be moved to one location (i.e., the python equivalent of a fortran module) and accessed from `create_map.py` and `create_plot.py` via the python equivalent a fortran "module use" statement?

File: src/emcpy/plots/create_plot.py
Comment (aerorahul) [inline src/emcpy/plots/create_plot.py:358]: @CoryMartin-NOAA and @aerorahul had a chat on this and even @RussTreadon-NOAA agrees that this duplicate code needs to be pulled out.
We should create an issue immediately and create a priority PR for it.

**Comment by CoryMartin-NOAA** [comment]
In fact, this was raised here https://github.com/NOAA-EMC/emcpy/issues/49. @aerorahul would it be preferred to tackle this before merging this PR or after?

**Comment by aerorahul** [comment]
> In fact, this was raised here #49. @aerorahul would it be preferred to tackle this before merging this PR or after?

your call @CoryMartin-NOAA 
I am fine with after this is merged so we can move on and get the figures like we want them.
The change would be transparent in the next PR.

**Review by aerorahul — APPROVED** [review]

**Comment by CoryMartin-NOAA** [comment]
perfect, thanks @aerorahul, I agree I think this should be merged and then refactor, since with the unit tests that is a good way to verify the changes are transparent to end users.

**Comment by kevindougherty-noaa** [comment]
Closes #43

**Pull request merged**
